src/openvic-simulation/economy/FactoryProducer.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/pop/Pop.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+#include "utility/Constructors.hpp"
+
+namespace OpenVic {
+	class FactoryProducer final {
+	private:
+		static constexpr uint8_t DAYS_OF_HISTORY = 7;
+
+	public:
+		typedef ordered_map<Pop* const, Pop::pop_size_t> employees_map_t;
+		typedef std::array<fixed_point_t, DAYS_OF_HISTORY> daily_profit_history_t;
+
+	private:
+		uint8_t PROPERTY(profit_history_current);
+		daily_profit_history_t PROPERTY(daily_profit_history);
+
+	public:
```

Comment:
`PROPERTY` [always creates `public` getter functions](https://github.com/OpenVicProject/OpenVic-Simulation/blob/master/src/openvic-simulation/utility/Getters.hpp#L109), then resets the accesibility state back to `private`. If the variables themselves are `public` then they can be accessed without the getter functions, and in particular they can be modified externally. All `PROPERTY` variables in this file and the others added in this PR should be `private`, `PROPERTY` will make sure the getters will still be `public`, and if a variable needs to be modified like `hiring_priority`, then it should use `PROPERTY_RW`.

Currently all the variables here apart from `production_type` are already `private`, because of `PROPERTY` setting the accesibility state to `private` after declaring the `public` getter function. The use of `PROPERTY_RW_ACCESS` for `hiring_priority` changes the accessibility status to `public` after the getter function, but not before, so `hiring_priority` is also already `private`. The `*_ACCESS` variants are meant for when the variable isn't `private` (so when it's `protected`), if you're changing the accessibility status to something different from the variable then that should be done with a regular `public` keyword to make it clear that the stuff following it has a different accessibility state.